cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -143,21 +144,31 @@ func (s *CMServer) AddFlags(fs *pflag.FlagSet) {
 	fs.IPNetVar(&s.ClusterCIDR, "cluster-cidr", s.ClusterCIDR, "CIDR Range for Pods in cluster.")
 	fs.BoolVar(&s.AllocateNodeCIDRs, "allocate-node-cidrs", false, "Should CIDRs for Pods be allocated and set on the cloud provider.")
 	fs.StringVar(&s.Master, "master", s.Master, "The address of the Kubernetes API server (overrides any value in kubeconfig)")
-	fs.StringVar(&s.Kubeconfig, "kubeconfig", s.Kubeconfig, "Path to kubeconfig file with authorization and master location information.")
+	fs.MarkDeprecated("master", "please use --api-servers instead")
+	fs.StringSliceVar(&s.APIServerList, "api-servers", s.APIServerList, "Comma separated list of api servers. (ex: http://host1:8080,https://host2:443). NOTE: Only the first server will be used")
+	fs.StringVar(&s.Kubeconfig, "kubeconfig", s.Kubeconfig, "Path to kubeconfig file with authorization and apiserver location information.")
 	fs.StringVar(&s.RootCAFile, "root-ca-file", s.RootCAFile, "If set, this root certificate authority will be included in service account's token secret. This must be a valid PEM-encoded CA bundle.")
 }
 
 // Run runs the CMServer.  This should never exit.
 func (s *CMServer) Run(_ []string) error {
-	if s.Kubeconfig == "" && s.Master == "" {
-		glog.Warningf("Neither --kubeconfig nor --master was specified.  Using default API client.  This might not work.")
+	if s.Master != "" {
+		if len(s.APIServerList) != 0 {
+			glog.Warningf("Both --mater and --api-servers set. Using --api-servers")
```

Comment:
Yeah!!! Do you have a black mage sticker in your laptop too?

![img_20160427_164031](https://cloud.githubusercontent.com/assets/729830/14865827/f9bc4f1a-0c96-11e6-9c9b-57207b752d82.jpg)